dist/manifestor.js

```diff
@@ -3534,14 +3550,17 @@ var manifestLayout = function(options) {
       if (!line) { line = this.addLine(); }
 
       if (line.remaining >= lineItemWidth) {
-        var x = lineWidth - line.remaining;
+        x = lineWidth - line.remaining;
         if (viewingDirection === 'right-to-left') {
           x = line.remaining - frame.x;
-        };
+        }
         line.remaining -= frame.width;
         return [x, lines.currentLine];
       }
-
+      if (line.remaining >= frame.width && facingPageType(frame.canvas.sequencePosition) === 'rightPage') {
+        x = lineWidth - line.remaining;
+        return [x, lines.currentLine];
+      }
```

Comment:
Not sure how this block is going to jive with the right-to-left code, but let's tackle that next.